File: leetcode_problems/python/lc441_arranging_coins.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        lo = 1
        hi = n
        while lo < hi:
            mid = lo + (hi - lo + 1) // 2
            c = mid * (mid + 1) // 2
            if c == n:
                return mid
            elif c > n:
                hi = mid - 1
            else:
                lo = mid

        return lo


if __name__ == "__main__":
    print(Solution().arrangeCoins(5))
    print(Solution().arrangeCoins(8))
    print(Solution().arrangeCoins(10))

File: leetcode_problems/cpp/lc441_arranging_coins.cpp
class Solution
{
public:
    bool is_good(long long mid, long long n)
    {
        long long sum = mid * (mid + 1) / 2;
        return sum <= n;
    }

    int arrangeCoins(int n)
    {
        long long low = 0;
        long long high = n + 1;
        while (high - low > 1)
        {
            long long mid = low + (high - low) / 2;
            if (is_good(mid, n))
            {
                low = mid;
            }
            else
            {
                high = mid;
            }
        }
        return low;
    }
};